depends/libhdfs3/src/client/HttpClient.h

```diff
@@ -0,0 +1,155 @@
+/********************************************************************
+ * 2014 -
+ * open source under Apache License Version 2.0
+ ********************************************************************/
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef _HDFS_LIBHDFS3_CLIENT_HTTPCLIENT_H_
+#define _HDFS_LIBHDFS3_CLIENT_HTTPCLIENT_H_
+
+#include <string>
+#include <vector>
+#include <curl/curl.h>
+#include "Exception.h"
+#include "ExceptionInternal.h"
+
+typedef enum httpMethod {
+    HTTP_GET = 0,
+	HTTP_POST = 1,
+	HTTP_DELETE = 2,
+	HTTP_PUT = 3
+} httpMethod;
+
+namespace Hdfs {
+
+class HttpClient {
+public:
```

Comment:
I think we should add a unit test for HttpClient, because libcurl have many tricky behaviors, maybe we forgot some exceptional path.
Just a reminder, let's do it in future.